src/scripts/modules/list.js

```diff
@@ -53,8 +57,30 @@ function renderDestination(destination) {
   const img = document.createElement("img");
   img.style.objectFit = "cover"; // TODO - Prefer adding a CSS class instead.
   img.src = destination.photo;
+  //create checkbox
+  const checkbox = document.createElement("input");
+  checkbox.type = "checkbox";
+  checkbox.classList.add("check");
+
+  // move checked items to 'visited' list
+
+  checkbox.addEventListener("click", (destination) => {
+    toggleButtonVisited(event, destination);
+  });
+
+  checkbox.addEventListener("click", (event) => {
+    toggleButtonVisited(event, destination);
+  });
```

Comment:
Oh it's a mistake, wasn't my intention to have two.  ```toggleButtonVisited(event, item) {....}``` doesn't need the event then? I don't get how this call works

Reply:
The event doesn't add anything useful in this case. If you want, you can pass the current toggle status for example

```js
toggleButtonVisited(destination.id, event.target.checked);
```

Then on the function itself, you receive both:

```js
export function toggleButtonVisited(itemId, itemChecked) {
  // Toggle the item visited status
  destinationsObj[itemId].visited = !itemChecked

  // ...
  // <Code to update local storage>...
  // ...

  // Pass the new destinations as a list to pubsub
  const newList = destinationsList()
  events.publish("destinationUpdate", newList);
}
````

Do you understand it better now? What exactly are you struggling to understand?

Reply:
P.S. Check my last comment, I updated it.

Reply:
What I understood is that toggleButtonVisited() should stay mostly as is so that's why I couldn't see it working, the way you put it here makes it clear.

Reply:
```    toggleButtonVisited(destination.id);```

Is there a specific reason why here just the id is passed instead of the whole item?

Reply:
I ask you the same question, is there a reason to pass the whole item? If yes, go for it. If not, keep it simple ;)

Reply:
Fair enough 😄